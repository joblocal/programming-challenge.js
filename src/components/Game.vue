<template>
  <div>
    <header>TicTacToe</header>

    <div class="Game">
      <Board />
    </div>

    <footer>
      &copy; Joblocal GmbH
    </footer>
  </div>
</template>

<script>
import Board from 'src/components/Board';

export default {
  components: {
    Board,
  },
};
</script>

<style>
:root {
  --header-height: 4rem;
  --footer-height: 2rem;

  --without-header-footer: var(--header-height) - var(--footer-height);
  --board-size: calc(100vmin - var(--without-header-footer));
}

body {
  margin: 0;
  padding: 0;

  background: #ccc;
}

header {
  line-height: var(--header-height);
  font-size: 3rem;
  text-align: center;
}

footer {
  line-height: var(--footer-height);
  text-align: center;
}

.Game {
  display: block;
  margin: 0 auto;

  height: var(--board-size);
  width: var(--board-size);
}
</style>
