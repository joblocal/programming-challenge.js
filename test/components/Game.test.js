import Vue from 'vue';

// module under test
import Game from 'src/components/Game';

test('matches snapshot', () => {
  const vm = new Vue(Game).$mount();
  expect(vm.$el).toMatchSnapshot();
});
