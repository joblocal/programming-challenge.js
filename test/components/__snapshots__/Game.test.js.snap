// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <header>
    TicTacToe
  </header>
   
  <div
    class="Game"
  >
    <div
      class="Board"
    >
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
      <div
        class="Field"
      />
    </div>
  </div>
   
  <footer>
    
    Â© Joblocal GmbH
  
  </footer>
</div>
`;
